Engine/ResultProxy.cs

```diff
@@ -632,7 +632,15 @@ public void Invoke(IResultListener a, WorkQueue queue)
             {
                 try
                 {
-                    a.OnResultPublished(proxy.stepRun.Id, CreateOptimizedTable(queue));
+                    // only merge result tables where it is explicitly supported by the result listener.
+                    if (a is IMergedTableResultListener m && m.SupportsMergedResults)
+                    {
+                        a.OnResultPublished(proxy.stepRun.Id, CreateOptimizedTable(queue));    
```

Comment:
Did you implement the SkippableInvokable logic because the call to `CreateOptimizedTable` is expensive? It is a bit unclear to me why we need to call `CreateOptimizedTable` for each result listener in the first place.